Composer/packages/client/src/pages/language-generation/index.tsx

```diff
@@ -64,6 +64,23 @@ const LGPage: React.FC<LGPageProps> = props => {
     }
   }, [fileId, dialogs, projectId]);
 
+  useEffect(() => {
+    const handleUserKeyPress = event => {
+      const { keyCode } = event;
+      //when press esc and in edit mode
+      if (keyCode === 27 && edit) {
+        const url = `/bot/${projectId}/language-generation/${fileId}`;
+        navigateTo(url);
+      }
+    };
+
+    document.addEventListener('keydown', handleUserKeyPress);
```

Comment:
I guess Andy means move EventListenerTarget from document to CodeEditor itself ,then pop up an event like ' editorWantClose ' somehow let page to handle it.

Reply:
@a-b-r-o-w-n  @zhixzhan moved eventHandler down to baseEditor

Reply:
This is not what I mean. I mean that instead of the document having the event listener, some element closer to the editor should have it. For example:

```ts
<div onKeydown={handleKeydown}>
  <CodeEditor {...props} />
</div>
```

I do not think this behavior belongs in the code-editor package.

Reply:
Another option here is to add a custom key command upon the editor mounting like in https://github.com/microsoft/BotFramework-Composer/blob/master/Composer/packages/lib/code-editor/src/LuEditor.tsx#L122

Reply:
applied option 2 now

Reply:
> I do not think this behavior belongs in the code-editor package.

See my comment below.